lib/charms/alertmanager_k8s/v0/alertmanager_remote_configurer.py

```diff
@@ -0,0 +1,1888 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Alertmanager Remote Configurer library.
+
+This library provides possibility of configuring the Alertmanager through another Juju charm.
+It has been created with the `alertmanager-k8s` and the `alertmanager-k8s-configurer` charms in
+mind, but can be used by any charms which require functionalities implemented by this library.
+
+Charms that need to push Alertmanager configuration to a charm exposing relation using
+the `alertmanager_remote_configurer` interface, should use
+the `AlertmanagerRemoteConfigurerConsumer`.
+Charms that need to can utilize the Alertmanager configuration provided from the external source
+through a relation using the `alertmanager_remote_configurer` interface, should use
+the `AlertmanagerRemoteConfigurerProvider`.
+
+For custom consumer implementations, two additional methods are available - `load_config_file` and
+`load_templates_file`. They can be used to prepare the relevant data to be pushed to the relation
+data bag.
+"""
+
+import json
+import logging
+import os
+from typing import Optional, Union
+
+import requests
+import yaml
+from jsonschema import exceptions, validate  # type: ignore[import]
+from ops.charm import CharmBase, RelationChangedEvent, RelationJoinedEvent
+from ops.framework import Object
+from ops.model import BlockedStatus, Relation
+
+# The unique Charmhub library identifier, never change it
+LIBID = "something dummy for now"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+
+ALERTMANAGER_CONFIG_JSON_SCHEMA = {
```

Comment:
Should we allow additionalProperties to make it backward compatible if the schema is ever extendeed?

Reply:
I thought about it and I guess it's a trade off. 
On one hand keeping the schema closed brings the need of maintaining it. On the other hand, opening it, means you can send pretty much anything you want and I'm not sure we want that either.
For now I've closed it, but I have no strong opinion here.

Reply:
The case I'm making for opening it is that if we decide to add a property in v2, any alertmanager using v2 will no longer be able to communicate with a remote charm using v1, even on a best-effort basis, as v1 will say "you have additional props here - DENIED!".

Reply:
Yes, I see your point.
@sed-i mentioned moving the whole schema to some other place outside of the library. I wonder if that would allow keeping the schema closed and validating against multiple schemes on the charm side at the same time?

Reply:
Schema removed, so I guess it doesn't matter anymore. 
I didn't know about the `amtool` (thanks @sed-i ;) )